### DISCUSSION TITLE
Can One Process have two spdlog instance？

### DISCUSSION THREAD

**Discussion by kanli201904** [gabime/spdlog#2435] [Q&A]
the problem is :The spdlog exception is  like : " the thread pool is not exist"
the case is :  We create one shared XXX.so  , that use spdlog to record some logs,   then one user use the XXX.so in his process.   the user also use the spdlog to record logs.  As the result: becaust the user also call the **spdlog::init_thread_pool**, that we destroy the spdlog's thread pool in the so.    the spdlog can not be generated inner the so module

**Comment by tt4g** [comment]
There is no support (#1448). But you can create thread-pool for each process.
See: https://github.com/gabime/spdlog/wiki/6.-Asynchronous-logging#spdlogs-thread-pool

**Reply by kanli201904** [reply]
@tt4g : thank you very much.  if we can' t use the theadpool in the XXX.so( only use the spdlog::rotating_logger_mt(),),   the urer process (which call the XXX.so) use the thread pool ,can the  solve the problem ?

**Reply by tt4g** [reply]
I think the problem will no longer occur.
Because The `spdlog::async_logger` uses a thread pool, but not `spdlog::rotating_logger_mt()` (that returns `spdlog::logger`). Therefore, `spdlog::init_thread_pool()` should not be called in XXX.so.

**Reply by kanli201904** [reply]
thanks, i have tried that , it works well , we need do more test
